esp-hal/src/mcpwm/operator.rs

```diff
@@ -316,7 +336,20 @@ impl<'d, Pin: OutputPin, PWM: PwmPeripheral, const OP: u8, const IS_A: bool>
         let dt_red = unsafe { Self::ch() }.db_red_cfg();
         #[cfg(not(esp32s3))]
         let dt_red = unsafe { Self::ch() }.dt_red_cfg();
-        dt_red.write(|w| unsafe { w.bits(cycles as u32) });
+        dt_red.write(|w| w.red().variant(cycles));
+    }
+
+    /// Updates dead-time CFG register
+    ///
+    /// WARNING: CFG is connected to the operator, and could be connected to
```

Comment:
It might be helpful to state why this could be bad, and what users can do to avoid that bad behaviour when updating the dead time?

Reply:
Or maybe something like:

```rust
impl Operator {
    pub fn with_linked_pins(self, pin_1: Pin1, config_a: PwmPinConfig, pin_2: Pin2, config_b: PwmPinConfig, config_dt: DeadTimeConfig) -> LinkedPins;
}

pub struct LinkedPins;

impl LinkedPins {
    pub fn set_actions_a(&mut self, value: PwmActions);
    pub fn set_actions_b(&mut self, value: PwmActions);
    pub fn set_update_method_a(&mut self, update_method: PwmUpdateMethod);
    pub fn set_update_method_b(&mut self, update_method: PwmUpdateMethod);
    pub fn set_timestamp_a(&mut self, value: u16);
    pub fn set_timestamp_b(&mut self, value: u16);
    pub fn set_deadtime_config(&mut self, config: DeadTimeConfig);
    pub fn set_rising_edge_deadtime(&mut self, dead_time: u16);
    pub fn set_falling_edge_deadtime(&mut self, dead_time: u16);
}
```

Reply:
I think the latter could work, but I will admit my knowledge of MCPWM is a bit limited, I need to do some reading :D. Maybe we could discuss this in a separate issue? I think the API for MCPWM could do with a bit of work regardless, but I don't want to block this PR.